api/src/pcapi/core/providers/titelive_apply_new_filters_on_history.py

```diff
@@ -0,0 +1,345 @@
+    gtl_id = kwargs.get("gtl_id")
+    code_support = kwargs.get("code_support")
+    id_lectorat = kwargs.get("id_lectorat")
+    taux_tva = kwargs.get("taux_tva")
+    title = kwargs.get("title")
+    comment = kwargs.get("comment")
+
+    gtl = None
+
+    if title:
+        title = title.lower()
+
+    if comment:
+        comment = comment.lower()
+
+    try:
+        if gtl_id:
+            gtl = get_gtl(gtl_id)
+    except KeyError:
+        pass
+
+    # Ouvrage avec pierres ou encens, jeux de société ou escape game en coffrets,
+    # marchandisage : jouets, goodies, peluches, posters, papeterie, etc...
+    if taux_tva == BASE_VAT:
+        return "vat-20"
+
+    # ouvrage du rayon scolaire
+    if gtl and gtl["level_01_code"] == GTL_LEVEL_01_SCHOOL:
+        return "school"
+
+    # ouvrage du rayon parascolaire,
+    # code de la route (méthode d'apprentissage + codes basiques), code nautique, code aviation, etc...
+    if gtl and gtl["level_01_code"] == GTL_LEVEL_01_EXTRACURRICULAR:
+        return "extracurricular"
+
+    if code_support == CALENDAR_SUPPORT_CODE:
+        return "calendar"
+
+    if code_support == POSTER_SUPPORT_CODE:
+        return "poster"
+
+    if code_support == PAPER_CONSUMABLE_SUPPORT_CODE:
+        return "paper-consumable"
+
+    # Coffrets (contenant un produit + un petit livret)
+    if code_support == BOX_SUPPORT_CODE:
+        return "box"
+
+    # Oracles contenant des jeux de tarot
+    if code_support == OBJECT_SUPPORT_CODE:
+        return "object"
+
+    # # ouvrage "lectorat 18+" (Pornographie / ultra-violence)
+    if id_lectorat == LECTORAT_EIGHTEEN_ID and comment and ADULT_ADVISOR_TEXT in comment:
+        return "pornography-or-violence"
+
+    # Petite jeunesse (livres pour le bains, peluches, puzzles, etc...)
+    if (
+        gtl
+        and gtl["level_01_code"] == GTL_LEVEL_01_YOUNG
+        and gtl["level_02_code"]
+        in [
+            GTL_LEVEL_02_BEFORE_3,
+            GTL_LEVEL_02_AFTER_3_AND_BEFORE_6,
+        ]
+    ):
+        return "small-young"
+
+    # Toeic or toefl
+    if TOEIC_TEXT in title or TOEFL_TEXT in title:
+        return "toeic-toefl"
+
+    if taux_tva == PAPER_PRESS_VAT and code_support == PAPER_PRESS_SUPPORT_CODE:
+        return "press"
+
+    return None
+
+
+def _get_remote_to_local_cmd(file_path: str, env: str) -> str:
+    return f"kubectl cp -n {env} {HOSTNAME}:{file_path} {os.path.basename(file_path)}"
+
+
+def update_products(file_path: str, comments_file_path: str, dry: bool) -> None:
+    product_whitelist_eans = {
+        ean for ean, in fraud_models.ProductWhitelist.query.with_entities(fraud_models.ProductWhitelist.ean).all()
+    }
+
+    comments_dict: dict[str, str] = {}
+
+    with open(comments_file_path, newline="", encoding="utf-8") as csv_comments_file:
+        csv_reader = csv.reader(csv_comments_file, delimiter=";")
+        for index, row in enumerate(csv_reader, HEADER_LINE_COUNT):  # pylint:disable=unused-variable
```

Comment:
(bug) On s'est mal compris à propos de mon commentaire sur `enumerate()`. Le 2ème argument d'`enumerate()` indique le premier index renvoyé. Exemple : 

```python
>>> list(enumerate(["a", "b", "c"]))
[(0, 'a'), (1, 'b'), (2, 'c')]
>>> list(enumerate(["a", "b", "c"], 1))
[(1, 'a'), (2, 'b'), (3, 'c')]
```

Donc, ici, la première valeur d'index sera 1, mais la boucle va itérer sur **toutes** les lignes du fichier CSV. Donc ça va mettre de mauvaises valeurs dans "ean" et "comment" à cause de la 1ère ligne qui contient le nom des colonnes.

La bonne façon de faire : 

```suggestion
        for index, row in enumerate(csv_reader, 1):
            if index <= HEADER_LINE_COUNT:
                continue
```

Idem dans la boucle plus bas qui lit l'autre fichier CSV.